pallets/parachain-staking/src/lib.rs

```diff
@@ -724,19 +847,52 @@ pub mod pallet {
 			Perbill,
 			Perbill,
 		),
+		/// Migrated NominatorState -> NominatorState2, ExitQueue -> ExitQueue2
+		DelayNominationExitsMigrationExecuted,
+	}
+
+	/// Storage migration for delaying nomination exits and revocations
+	fn delay_nomination_exits_migration_execution<T: Config>() {
+		if !<DelayNominationExitsMigration<T>>::get() {
+			// migrate from Nominator -> Nominator2
+			for (acc, nominator_state) in NominatorState::<T>::drain() {
+				let state: Nominator2<T::AccountId, BalanceOf<T>> = nominator_state.into();
+				<NominatorState2<T>>::insert(acc, state);
+			}
+			// migrate from ExitQueue -> ExitQueue2
+			let just_collators_exit_queue = <ExitQueue<T>>::take();
+			let mut candidates: Vec<T::AccountId> = Vec::new();
+			for (acc, _) in just_collators_exit_queue.clone().into_iter() {
+				candidates.push(acc);
+			}
+			<ExitQueue2<T>>::put(ExitQ {
+				candidates: candidates.into(),
+				nominators_leaving: OrderedSet::new(),
+				candidate_schedule: just_collators_exit_queue,
+				nominator_schedule: Vec::new(),
+			});
+			<DelayNominationExitsMigration<T>>::put(true);
+			Pallet::<T>::deposit_event(Event::DelayNominationExitsMigrationExecuted);
+		}
```

Comment:
@notlesh this is the full migration logic

It emits an event when it executes and also replaces a false storage value with true so it doesn't run more than once.